File: app/src/main/java/com/robotnec/budget/core/service/aggregation/impl/TransactionAggregation.java
package com.robotnec.budget.core.service.aggregation.impl;

import com.robotnec.budget.core.domain.operation.Transaction;

import java.util.List;
import java.util.SortedMap;

/**
 * @author zak <zak@swingpulse.com>
 */
public interface TransactionAggregation {
    int getSpansCount();

    SortedMap<TimeSpan, List<Transaction>> get(Sorting sorting);

    enum Sorting {
        ASC, DESC
    }
}

File: app/src/main/java/com/robotnec/budget/core/service/aggregation/impl/TransactionAggregation.kt
package com.robotnec.budget.core.service.aggregation.impl

import com.robotnec.budget.core.domain.operation.Transaction
import org.joda.money.Money
import java.util.SortedMap

/**
 * @author zak zak@swingpulse.com>
 */
interface TransactionAggregation {
    val spansCount: Int

    operator fun get(sorting: Sorting): SortedMap<TimeSpan, List<Transaction>>

    fun getSum(span: TimeSpan): Money

    enum class Sorting {
        ASC, DESC
    }
}
